accepted/0000-hub_retail.md

```diff
@@ -0,0 +1,310 @@
+
+In long term, alternative 2 is clean solution.
+
+Alternative 1 might be used as a short term fix and also to keep compatibility.
+
+Note:
+The Formula pillar files depend on group IDs. The IDs differs between Hub and Peripheral server so
+direct file-based synchronization of pillars is not possible.
+
+
+YAML configuration
+------------------
+
+FIXME: optional, related to Alternative 2 in previous section
+
+We have retail_yaml tool for initial configuration of Retail environment from
+[YAML](https://github.com/uyuni-project/retail/blob/master/python-susemanager-retail/example.yml) file.
+[Hub Users RFC](00061-hub-users.md) introduced similar concept, based on Salt.
+Maybe these parts could be merged.
+
+
+Combined Peripheral and Branch server
+-------------------------------------
+
+Branch services are configured by Salt formulas. Also PXE entries for each terminal are generated
+by Salt (currently via reactor).
+With combined Peripheral and Branch server, these formulas should be applied on SUMA server itself.
+
+Terminals connect to SUMA directly, SUMA Proxy is not needed.
+
+
+FIXME: pick one alternative
+
+Alternative 1: Hub as a Salt Master for Combined Peripheral and Branch server
+
+- this is already used for [Hub Users RFC](00061-hub-users.md)
+
+problems:
+- does not work for PXE entries https://github.com/uyuni-project/uyuni-rfc/pull/46
+  The event from each booting Terminal must be stopped on Peripheral server, on
+  Hub it would cause the same scalability problems as before.
+
+- mixed environment with Hub, Peripheral Servers with standalone Branch Serveres and
+  Combined Peripheral Servers will have non-optimal configuration
+  For example: A shop uses Combined Peripheral and Branch server, then the number of terminals grows and
+  at some point it is necessary to add standalone Branch Server. Then the configuration of the
+  shop would be split between Hub and Peripheral Server, or the shop would have to be reinstalled
+  with 2 Branch Servers and standalone Peripheral Server
+
+- does not cover the "Single server" setup
+
+
+Alternative 2: Multi-master setup of Combined Peripheral and Branch server
+
+- Salt minion running on Combined Peripheral and Branch server has 2 masters - Hub and itself
+- the Branch Server configuration via formulas can stay unchanged
+
+problems:
+- SUMA registered to itself cause many problems
+  - we already have clients that do not allow certain actions (containers) so this should be fixable
+  - if we allow and test this scenario for Retail, we will know how it behaves for
+  non-Retail use and the most severe problems will be fixed (instead of just saying "we don't know")
+- possible security implications - users will gain access to the server
+  - this is not a problem for Retail. Peripheral server will be managed by Hub admins who can access everything
+  and optionally by local admins who should access for example all machines in one building.
+  There is no need for other roles.
```

Comment:
This is an assumption that we know how our users use the product with which I can't agree.

Given the possibilities of Uyuni I am pretty sure users would want to use peripheral server for more then just 'offline capable proxy', so management of terminals, but also management of servers running local database instance for retail app and what not.
For that it is easy to imagine they have different users with different roles, e.g. one junior admin assigned to branch group maintaining terminals with low level of risk. Then different admin, more senior posion, assigned to group maintaining backend services.
Now if this user follow recommendation wrt proxy we currently have (and retail_yaml does automatically), they may be tempted to add this peripheral server to the group with terminals as we do with branch proxy. At this moment the junior admin maintaining terminals will get root access to the peripheral server itself and thus root access to backend servers as well.

Reply:
I don't think that one small shop (which needs this combined server to save resources) will need all these admin roles.

Reply:
I am sorry, but this is not just one small shop. This is scenario with hub and combined peripheral server. This would be every user who want to have offline functionality.

But at the end all of this is speculation. I just really don't like statements 'There is no need for other roles.' IT is plenty of similar statements only to be proved wrong. Thus I don't agree with using these statements as a reasoning for some decision.